addons/auto_index/models/missing_indexes.py

```diff
@@ -0,0 +1,138 @@
+    idx_scan = fields.Integer(string="Number of scans", readonly=True)
+    idx_tup_read = fields.Integer(string="Number of tuple returned", readonly=True)
+    idx_tup_fetch = fields.Integer(string="Number of fetches", readonly=True)
+    index_definition = fields.Char(string="Index definition", readonly=True)
+    index_size = fields.Char(string="Index size on disk", readonly=True)
+    index_size_raw = fields.Integer(string="Index size on disk in bytes", readonly=True)
+    is_unique = fields.Boolean(string="Is unique", readonly=True)
+    is_clustered = fields.Boolean(string="Is clustered", readonly=True)
+    is_primary = fields.Boolean(string="Is Primary", readonly=True)
+
+    def init(self) -> None:
+        tools.drop_view_if_exists(self._cr, self._table)
+        self._cr.execute(f"""
+CREATE OR REPLACE VIEW {self._table} as
+SELECT
+        row_number() OVER() AS id,
+        relid                                                    as table_id,
+        pi2.indexrelid                                           as index_id,
+        relname                                                  as table_name,
+        indexrelname                                             as index_name,
+        idx_scan,
+        idx_tup_read,
+        idx_tup_fetch,
+        indexdef                                                 as index_definition,
+        pg_size_pretty(pg_table_size(stat_all_index.indexrelid)) as index_size,
+        pg_table_size(stat_all_index.indexrelid)                 as index_size_raw,
+        pi2.indisunique                                          as is_unique,
+        pi2.indisclustered                                       as is_clustered,
+        pi2.indisprimary                                         as is_primary
+FROM pg_catalog.pg_stat_all_indexes stat_all_index
+         INNER JOIN pg_catalog.pg_indexes pi ON stat_all_index.indexrelname = pi.indexname
+         INNER JOIN pg_catalog.pg_index pi2 ON pi2.indexrelid = stat_all_index.indexrelid
+WHERE pi.schemaname <> 'pg_catalog'
+        """)
+
+    def remove_index(self):
+        for index in self:
+            tools.drop_index(self.env.cr, index.index_name, index.table_name)
+            self.env["missing.index_log"].create(
+                {"index_command": index.index_definition, "action": "removed", "index_name": index.index_name})
+
+    def remove_all_unused_indexes(self):
+        all_unused_indexes = self.search(domain=[('idx_scan', '=', 0), ('idx_tup_read', '=', 0), ('idx_tup_fetch', '=', 0), ('is_unique', '=', False), ('is_primary', '=', False), ('is_clustered', '=', False)])
+
+        all_impacted_tables = OrderedSet(all_unused_indexes.mapped('table_name'))
+
+        all_unused_indexes.remove_index()
+
+        for impacted_table in all_impacted_tables:
+            # pylint: disable=sql-injection
+            # input doesn't come from the user
+            self.env.cr.execute(f'analyse {impacted_table};')
```

Comment:
Alternative version:
```suggestion
            # input doesn't come from the user
            self.env.cr.execute(sql.SQL('analyse {};').format(sql.Identifier(impacted_table)))
```

Reply:
It'll also work correctly if the `impacted_table` must be quoted, which can happen.